overlord/devicestate/devicemgr.go

```diff
@@ -554,6 +554,37 @@ func (m *DeviceManager) ensureInstalled() error {
 	return nil
 }
 
+var timeNow = time.Now
+
+// StartOfOperationTime returns the time when snapd started operating,
+// and sets it in the state when called for the first time.
+// The StartOfOperationTime time is seed-time if available,
+// or current time otherwise.
+func (m *DeviceManager) StartOfOperationTime() (time.Time, error) {
```

Comment:
I forgot one thing, in preseed mode this should not set something in state but just return time.Time{} or maybe even return an error?

Reply:
Ok, I was planning to simply avoid Prune at the overlord level, but it's a good idea to error out here in preseed mode in case this method gets called elsewhere in the future, will do.